tests/labs/lie/functional/common.py

```diff
@@ -122,17 +147,52 @@ def check_lie_group_function(module, op_name: str, atol: float, inputs, funcs=No
                     func(jac_impl), func(jac), atol=atol, rtol=atol
                 )
 
+    # Check multi-batch consistency
+    if batch_size is None:
+        return
+    lb = len(batch_size)
+    flattened_inputs = [x.reshape(-1, *x.shape[lb:]) for x in inputs]
+    out = op(*inputs)
+    flattened_out = op(*flattened_inputs)
+    if jop is None:
+        return
+    jout = jop(*inputs)[0]
+    flattened_jout = jop(*flattened_inputs)[0]
+    torch.testing.assert_close(out, flattened_out.reshape(*batch_size, *out.shape[lb:]))
+    for j, jf in zip(jout, flattened_jout):
+        torch.testing.assert_close(j, jf.reshape(*batch_size, *j.shape[lb:]))
 
-def left_project_func(module, group):
-    sels = range(group.shape[0])
 
+def left_project_func(module, group, batch_dim):
     def func(matrix: torch.Tensor):
-        return module._left_project_autograd_fn(group, matrix[sels, ..., sels, :, :])
+        assert matrix.ndim == 2 * batch_dim + 3  # shape should be (*BD, f, *BD, g1, g2)
+        g = group.clone()
+        # Convert to single-batch-dim sparse gradient format
+        batch_size = matrix.shape[:batch_dim]
+        if batch_dim > 0:
+            d = reduce(lambda x, y: x * y, batch_size)
+            matrix = matrix.reshape(d, -1, d, *group.shape[-2:])
+            sels = range(matrix.shape[0])
+            matrix = matrix[sels, ..., sels, :, :]
+            g = group.reshape(d, *group.shape[-2:])
+        # Compute projected gradient matrix
+        ret = module._left_project_autograd_fn(g, matrix)
```

Comment:
Do we assume the same dimension of inputs and outputs?

Reply:
Yes, I think so. There should be no broadcasting in the test where this function is used.